types/mochawesome/addContext/index.d.ts

```diff
@@ -0,0 +1,36 @@
+import * as Mocha from 'mocha';
+
+interface TestContextObject {
+    title: string;
+    value: any;
+}
+
+type TestContext = string | TestContextObject;
+
+/**
+ * Add context to the test object so it can
+ * be displayed in the mochawesome report.
+ *
+ * **Note**: Arrow function tests are not currently
+ * supported, since we depend on the current test
+ * context (i.e. `this`)
+ *
+ * @example
+ * it('should test something', function () {
+ *   someFunctionThatTestsCode();
+ *
+ *   addContext(this, 'some context to add');
+ *
+ *   addContext(this, {
+ *     title: 'Expected number of something'
+ *     value: 42
+ *   });
+ *
+ *   assert('something');
+ * });
+ *
+ */
+declare function addContext(testCtx: Mocha.Context, context: TestContext): void;
+declare namespace addContext { }
+
+export = addContext;
```

Comment:
This file should be at `addContext.d.ts`, not `addContext/index.d.ts`: https://unpkg.com/browse/mochawesome@6.2.2/src/

Reply:
Will refactoring this file to your suggestion allow it to continue working via the following import syntax? If so, I'll most definitely update it.
```ts
import * as addContext from 'mochawesome/addContext'
```

Reply:
Yes. Moreover, users complying with the actual ESM spec (e.g., those running with actual ESM support in Node 14+ need to include file extensions, and do not get any special meaning from files named `index`, so the correct import for those users is `'mochawesome/addContext.js'`, which is currently broken with the file structure in the PR now. The TLDR is that your job as a DT author is to represent the structure of the files that actually exist in the JS library. If you do that, imports will work for everyone. On the other hand, if you think you know what the imports are _supposed_ to look like and use that to guide your DT contribution, you’re very likely to make imports work for you, but be broken for other people. We can thank JavaScript’s fractured module ecosystem for that. 🙃

Reply:
Alright, well, considering that, it may not be worth further review from you since I'll need to do a bit of switching around since `import * as addContext` is not the correct way to be importing this in the first place.

Reply:
You’re already close! Once you get a chance to make these updates, I’ll look at the JSDoc issue with the namespace if it doesn’t sort itself out in the process.